### PR TITLE
working folder for group collab

### PR DESCRIPTION
**Pull request by xibz**
Session #1 

@sbrennan4

### CODE REVIEW

File: ticket-proposal/tickets_idea_and_questions.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+
+So, maybe this approach would not work, since system would need to be aware of
+tickets which I do not know of any system that does this today. However, some
+standard need to be introduced to persist that ticketing metadata.
+
+## Ticket States
+
+* Created/Open
+* InProgress (from the development point of view)
+* Reviewed
```
Comment (sbrennan4) [inline ticket-proposal/tickets_idea_and_questions.md:100]:
```suggestion
* Peer Review
```

File: ticket-proposal/tickets_idea_and_questions.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+* Approval statuses
+    - approval.declined
+    - approval.inprogress
+    - approval.approved
+* Testing
```
Comment (sbrennan4) [inline ticket-proposal/tickets_idea_and_questions.md:105]:
```suggestion
* QA Testing
```

File: ticket-proposal/tickets_idea_and_questions.md [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+## Ticket States
+
+* Created/Open
+* InProgress (from the development point of view)
+* Reviewed
+* Approval statuses
+    - approval.declined
+    - approval.inprogress
+    - approval.approved
```
Comment (sbrennan4) [inline ticket-proposal/tickets_idea_and_questions.md:104]: Maybe we don't need these sub-states.

```suggestion
* Approval/Sign-off
* Ready to Deploy
```

`Approval` meaning it is waiting for approval. If approved, it goes to either `Ready to Deploy` or `Testing` or some other state depending on the team's workflow. If rejected, it goes back to `Open` or `In progress` or some other state.

File: ticket-proposal/tickets_idea_and_questions.md [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+* Testing
+* Resolved
```
Comment (sbrennan4) [inline ticket-proposal/tickets_idea_and_questions.md:106]: Should we include a state for `Deployment`? Could be required depending on the definition of done.

```suggestion
* Deployment
* Resolved
```

Comment (sbrennan4) [inline ticket-proposal/tickets_idea_and_questions.md:106]:
```suggestion
* Resolved/Done
```

**Review by sbrennan4 — COMMENTED** [review]
Some thoughts on ticket states. I think it makes the most sense to have events when a ticket transitions _to_ a state and not necessarily when a ticket transitions _from_ a state so we should update our naming based on that perspective and use the present tense.

Edit: Thinking about this further. The CDEvents vocabulary uses past tense for the predicates so if we're trying to map these states to predicates then we should keep that in mind.

File: ticket-proposal/tickets_idea_and_questions.md (hunk 2 above)
Comment (xibz) [reply ticket-proposal/tickets_idea_and_questions.md:105]: Why do we need to make a distinction between `Testing` and `QA Testing`?

**Comment by xibz** [comment]
> Thinking about this further. The CDEvents vocabulary uses past tense for the predicates so if we're trying to map these states to predicates then we should keep that in mind.

CDEvents are declarative, so it's whatever occurred from some service. However,. that doesn't we can't use present tense here, since what did occur was moving to that state. So I think changing them to be present tense makes sense

**Comment by xibz** [comment]
@sbrennan4 from SIG meeting CDEvents 10/23/2023

Ticket should have predicates: created, updated, and resolved
Updated with state free form

Have stringed enums for the `updated` field, but also allow it to be freeform:
```go
type TicketState string

const (
    Created TicketState = "created"
    InProgress TicketState = "created"
    Closed TicketState = "closed"
)

ticketEvent := ticket.NewTicket(ticket.Created) // ticket.NewTicket("something-custom")
```

```cdevents
{
    "subject": {
        "state": "created" or "something-custom"
    }
}
```

Approval may have score to indicate whether or not a manual approval is needed

**Comment by xibz** [comment]
Link to ticket proposal https://hackmd.io/@Mev468KqSlSu2GZpMzXKhQ/ByYuaEufa

**Comment by sbrennan4** [comment]
TODO: go over jira api and make sure everything fits into the spec

**Pull request closed without merging**